AdvancedTeleport-Bukkit/src/main/java/io/github/niestrat99/advancedteleport/commands/home/HomesCommand.java

```diff
@@ -95,77 +95,85 @@ public boolean onCommand(
+                        // like a pair in java?
+                        .map(
+                                pair -> {
+                                    final var home = (Home) pair[0];
+                                    final var canAccess = (boolean) pair[1];
+                                    final var baseComponent =
+                                            Component.text(home.getName())
+                                                    .hoverEvent(
+                                                            CustomMessages
+                                                                    .locationBasedTooltip(
+                                                                            sender,
+                                                                            home,
+                                                                            "homes"));
+
+                                    CoreClass.debug("Processing home: " + home.getName() + ", " + canAccess);
+
+                                    if (!canAccess) {
+                                        if (!MainConfig.get()
+                                                .HIDE_HOMES_IF_DENIED
+                                                .get())
+                                            return Component
+                                                    .empty(); // TODO: Make sure
+                                        // this doesn't cause
+                                        // an extra comma.
+                                        return baseComponent
+                                                .color(NamedTextColor.GRAY)
+                                                .decorate(TextDecoration.ITALIC);
+                                    }
+
+                                    return baseComponent.clickEvent(
+                                            ClickEvent.runCommand(
+                                                    "/advancedteleport:home "
+                                                            + (sender == target
+                                                            ? ""
+                                                            : target
+                                                            .getName()
+                                                            + " ")
+                                                            + home.getName()));
+                                })
+                        .toList());
+        CoreClass.debug("Message text built.");
 
         if (sender == target
                 && atPlayer instanceof ATFloodgatePlayer atFloodgatePlayer
                 && MainConfig.get().USE_FLOODGATE_FORMS.get()) {
             CoreClass.debug("Sender is being sent a homes form.");
-            atFloodgatePlayer.sendHomeForm();
+
+            if (!body.content().isEmpty() || !body.children().isEmpty()) {
```

Comment:
More reliable check here may just be checking if the homes collection is empty, rather than if the component is empty